code/_onclick/click.dm

```diff
@@ -446,7 +446,12 @@
 
 /// MouseWheelOn
 /mob/proc/MouseWheelOn(atom/A, delta_x, delta_y, params)
-	return
+	if(incapacitated(ignore_restraints = TRUE, ignore_stasis = TRUE))
+		return
+
+	var/obj/item/I = get_active_held_item()
+	if (I)
+		SEND_SIGNAL(I, COMSIG_MOUSE_SCROLL, src, delta_x, delta_y, params)
```

Comment:
The signal should be sent by the mob, and the held item should be listening to the mob, instead.
You can set up that listening on equipping and unequipping.

Reply:
alright, Ill check on that in a bit.

Reply:
okay, I assume what I did is something along the lines of what you wanted.